kucher/view/main_window/register_view_widget/dialog_export.py

```diff
@@ -0,0 +1,197 @@
+            return
+        try:
+            self._file = open(self.fileName, 'r')
+        except IOError:
+            _logger.exception(f'File {self.fileName} could not be open')
+            return
+
+        # Check before writing any register if they are available on the device and within the permitted range
+        self.imported_registers = yaml.safe_load(self._file)
+        try:
+            for reg_check in self.registers:
+                if reg_check.name in self.imported_registers.keys():
+                    if not reg_check.mutable:
+                        show_error('Import failed',
+                                   f'Cannot import {self.fileName}',
+                                   f'This parameter cannot be modified on this device:\n{reg_check.name}',
+                                   self.parent)
+                        self._file.close()
+                        _logger.exception(f'Import failed : This parameter cannot be modified on this '
+                                          f'device: {reg_check.name}')
+                        return
+
+                    elif reg_check.has_min_and_max_values and reg_check.type_id != Register.ValueType.BOOLEAN:
+                        if not (reg_check.min_value <= self.imported_registers[reg_check.name] <= reg_check.max_value):
+                            show_error('Import failed',
+                                       f'Cannot import {self.fileName}',
+                                       f'This parameter value is outside permitted range:\n{reg_check.name}',
+                                       self.parent)
+                            self._file.close()
+                            _logger.exception('Import failed :This parameter value is outside '
+                                              f'permitted range: {reg_check.name}')
+                            return
+
+        except Exception as ex:
+            show_error('Import failed',
+                       f'Cannot import {self.fileName}',
+                       f'Reason: {str(ex)}',
+                       self.parent)
+            _logger.exception(f'Could not write registers: {ex!r}')
+            self._file.close()
+            return
+
+        self._file.close()
+
+        async def executor():
+            for _attempt in range(3):
+                try:
+                    await self._writeRegisters()
+                    break
+                except Exception as ex:
+                    _logger.exception(f'Registers could not be loaded (attempt {_attempt + 1}/3')
+
+        asyncio.get_event_loop().create_task(executor())
+
+    async def _writeRegisters(self):
+        self._unwritten_registers = []
+        for reg in self.registers:
+            if reg.mutable and (reg.name in self.imported_registers.keys()):
+
+                try:
+                    value = self.imported_registers[reg.name]
+                    await reg.write_through(value)
+
+                except KeyError:
```

Comment:
This is incorrect. As I wrote in #30:

> If the file does not specify the value for a register, the old value of the register is left unchanged.

I.e., if the value is not provided for a register, just don't touch it. In the extreme case of an empty config file, the logic would just do nothing and report success.